content/docs/user-guide/how-to/add-deps-or-outs-to-a-stage.md

```diff
@@ -0,0 +1,52 @@
+# Add Dependencies or Outputs to a Stage
+
+There are situations where we have executed a stage, but later notice that some
+of the files/directories used by the stage as dependencies or created as outputs
+are missing from `dvc.yaml`.
+
+To add existing files/directories as <abbr>dependencies</abbr> or
+<abbr>outputs</abbr> to a stage, either edit the `dvc.yaml` file or use
+`dvc run` with the `--no-exec` option. Then use `dvc commit` to save the
+output(s) to the <abbr>cache</abbr> (and update `dvc.lock`).
```

Comment:
WDYT @shcheklein ? It's still specific to existing dep/out files or dirs. (which was our original intention). Do you think we need to generalize it even more and leave the `commit` part as an optional step?

Either way I say this is ready to be merged for now, but since I opened the PR, I can't approve it. If you agree, please do.

Reply:
> It's still specific to existing dep/out files or dirs (which was our original intention)

then name should be different?

the point is- name is general, content is not. it makes a bit misleading. Like people might think that this the only way to add, or that it's the most general way when add is needed, etc, etc

Reply:
OK so @imhardikj, to generalize this we need to:
- Invert the order of the paragraphs above (don't talk about existing files in the first one)
- The `dvc commit` part should be in the 2nd one
- Edit text accordingly so it makes sense after the change ^
- The example will also need to emphasize that the last part is needed only for existing deps/outs files or dirs ("Finally, we need...")

Reply:
@jorgeorpinel updated with changes as per your suggestion.

Reply:
Hmmm yeah it didn't make much sense. Bullets 3 and 4 weren't really addressed. I'm going to take over this part... Thanks, resolving.